addons/stock_delivery/wizard/stock_return_picking.py

```diff
@@ -7,10 +7,10 @@ class StockReturnPicking(models.TransientModel):
     _inherit = 'stock.return.picking'
 
     def _create_returns(self):
-        # Prevent copy of the carrier and carrier price when generating return picking
-        # (we have no integration of returns for now)
+        # Prevent copy of the carrier and carrier price when generating return picking if it does not support them.
         new_picking, pick_type_id = super()._create_returns()
         picking = self.env['stock.picking'].browse(new_picking)
-        picking.write({'carrier_id': False,
-                       'carrier_price': 0.0})
+        if picking.carrier_id and not picking.carrier_id.can_generate_return:
+            picking.write({'carrier_id': False,
+                           'carrier_price': 0.0})
```

Comment:
I got the idea but it could break the flow for other carrier since they handle the return differently. You should adapt starship to it or extract the condition to a sub method that will be override in starship